app/assets/stylesheets/automate_import_export.css

```diff
@@ -1,11 +1,11 @@
 .import-label {
-  margin: 20px 0 0 0;
+  margin: 20px 0 0;
```

Comment:
This looks like a bad rule.

Yes, `x 0 y 0` is technically equivalent to `x 0 y`, but feels like providing 4 dimensions is much clearer than randomly omitting one.

Reply:
would you turn https://github.com/stylelint/stylelint/tree/master/lib/rules/shorthand-property-no-redundant-values rule for all (margin, padding, border-color, border-radius, border-style, border-width, grid-gap)

Reply:
In the ideal world, I'd do a PR against it to add a config option... the only problem with this rule is the "x Y z Y" case.

I mean, "1 0 1 0" **should** become "1 0", "1 1 1 1" **should** become "1". But "1 0 2 0" should not become "1 0 2".

Reply:
would a solution where we comment something like
```css
...{
margin: 20px 0 0 0;  /* stylelint-disable-line */
}
```
be acceptable?
TBH: since this is auto fixed role I personally don't mind so much that its 3 or 4 attributes.

Reply:
I'm not a big fan of adding random comments just to skip a lint check. 

I *am* a fan of ignoring warnings we know are wrong :).

Reply:
Ignoring issues imho is the wrong approach, people get used to ignoring things, exception are made and over time quality is reduced. IMHO linting is green or red, if its green, than you can move ahead, if not, the PR is blocked.

Overall, these standard linting code seems to have rational behind them, I don't think we should auto accept other people linting rules, but on the other hand, not having linting (or not fully enforcing it) does not help either.

Personally, I prefer the short notation for both colors and padding, since they are linting rules, you would always have people not happy about all of the rules, in my eyes, enforcing it is still valuable and voids bike shedding.

Reply:
> Frankly I think ignoring issues is the honest approach :). I have yet to see a non-trivial project with 0 style issues where this is not caused simply by sweeping the worst offenders under the rug with magic comments.

I think while my PR address many of the issues, the point of this is (any similar tools like eslint, pretiier etc) is to ensure that new code quality is higher, new code should not be accepted with "issues" in it.


> Spend a while reviewing actual manageiq PRs and see for yourself how many special rules etc we would need to have to be able to do an automatic red=bad, green=good aproach ;). It's simply not realistic, the best thing we can expect from linters is some hint where to look, but we will always have to review the red ones one by one to see what makes sense and what doesn't.

I have been trying to do so over the last few weeks, and I would love to see those examples you refer to, to be clear - I am not saying that linter are going to replace the reviewer work - they simply tell the author what he needs to fix even prior to a review process.

> Don't take me wrong, of course I agree that we want to encourage people to fix linting issues. But the simple fact is, a linter goes by simple-to-simplistic rules, and sometimes the devel simply knows better. And once you start requiring all code to be linting issues free, that's the time the project has died the dead of bureaucracy :) (because suddenly the rules matter more than the people who actually work on the project).

The way I look at it is simple - use tools to automate common tasks, really - you don't want to comment about obvious stuff. in reality - the author will usually respond quickly to comments if he is still working on the patch, if it takes a long time to review and only then we start talking about style vs the actual change, its much more "expansive" in terms of context switches and less fun for all parties.

> As for the specific notation, I completely agree that the shorter the better.. But I would put consistency over this preference 100% time. So, if we have a one color which can be in the short notation and 2 more which can't .. well, consistency wins.

As I said earlier, different people will have different opinions when it comes to linting.. it will be hard to make everybody happy all the time, but I would really like to stop bike shedding :-)

Reply:
>  new code should not be accepted with "issues" in it

Completely agreed, as long as people have a say in the process of determining what constitutes an issue and what not :). (And that's really the only problem I have with your approach, as long as we keep it clear that the linter does **not** have the last word, we're completely agreed :).)

(And this is really about the approach in general, not related to CSS where none of this really matters given the small number of people touching it.)


> I would love to see those examples you refer to

Uh, not sure if I can give you an exhaustive list right now but to name a few:

  * all the toolbar definition code is using a different formatting simply to be readable given the hierarchy

  * any time a line is too long but a bit but would be less readable when split

  * any code that was just moved elsewhere and should be kept unchanged except for actual semantic changes, and refactored in a separate PR

  * any code which is special in that it does not fit into the normal categories in the project

    * we have a warning about using `console.log`, but ignore it in code that exists for debugging purposes 

    * we also have a different style for webpack config, simply to make upgrading webpacker possible

   * any time the linter rules is useful but does not handle an edge case well (for example, the margin rule is right for lenghts=1,2,4, just not for length=3, but that does not mean it's not a useful hint, just one I want to ignore sometimes)

I'm sure there's more :).


> they simply tell the author what he needs to fix even prior to a review process ... use tools to automate common tasks .. stop bike shedding

Yup, that way, it's a useful tool (as long as it doesn't spam :)), I do agree we want this, even for CSS :).